### ISSUE TITLE
Please support SQL Server FILESTREAM and FileTables

### ISSUE THREAD

**Issue by MarcoLoetscher** [aspnet/EntityFramework#4573]
Hi,
It would be nice if you support SQL Server FILESTREAM and FileTables in EF Core.

**Comment by ErikEJ** [comment]
How would that be different than the current support for varbinary(max) ?

**Comment by ArieJones** [comment]
I think it is more accurate to say that it would be beneficial to have streaming access support for FILESTREAM. This is what the Win32 streaming support is specifically for. Otherwise you can pull data from a FILESTREAM object as a varbinary(MAX) as @ErikEJ  mentioned. The streaming is not that good depending on the size of the object...but hey, ya do what you have to do. 

The more problematic aspect if you want to support it is how do you make this consistent with the rest of EF experience which is traditionally a two-way street. Meaning that you could theoretically easily map an existing FILESTREAM field but how would you be able to effectively create a new table with a FILESTREAM column via migrations? There are sticky prereqs that come with it .... 

How do you enable FILESTREAM if it is not enabled already(it is an instance wide setting) ?
How do you enforce the NON NULL unique row id requirement? 
How do you create the FILESTREAM filegroup? 

There are a couple of very manual ways that you can get around this currently in EF 6. However, they don't address the fact that you need to have a FILESTREAM enabled instance which requires you to change configuration and have specific objects set up within the target database to hold the FILESTREAM data. 

With that said, I would also advise that the FILETABLE support be split out into a separate issue, because although it piggybacks somewhat on the FILESTREAM architecture, it is inherently different and comes with it's own unique set of "features". 

So while I would openly support such a set of features, I would be estatic just to have a one way method of streaming the data using the Win32 processes through a property on a POCO.

**Comment by divega** [comment]
Agreed with previous comments that FileTable is distinct feature in SQL Server that would require extra work to support. I won't create a separate issue right now but we can do it when get to implement FILESTREAM support.

**Comment by sepplK** [comment]
Hi, is there any workarround for the missing SqlFileStream in windows?

**Comment by evil-shrike** [comment]
As I understand FILESTREAM support is missing due to the absence of SqlFileStream support in SqlClient on .NET Core. 
See dotnet/corefx#15652

**Comment by HashSix** [comment]
Is there any work around with regards to this I see that you can't even access the API through .net standard.

**Comment by pantonis** [comment]
Any news on this?

**Comment by ajcvickers** [comment]
@pantonis This issue is in the Backlog milestone. This means that it is not planned for the next release (EF Core 5.0). We will [re-assess the backlog](https://docs.microsoft.com/ef/core/what-is-new/release-planning) following the this release and consider this item at that time. However, keep in mind that there are many other high priority features with which it will be competing for resources.

**Comment by nielsvissers** [comment]
is there any news on this?

**Comment by ajcvickers** [comment]
@nielsvissers  This issue is in the Backlog milestone. This means that it is not planned for the next release (EF Core 6.0). We will [re-assess the backlog](https://docs.microsoft.com/ef/core/what-is-new/release-planning) following the this release and consider this item at that time. However, keep in mind that there are many other high priority features with which it will be competing for resources. Make sure to vote (👍) for this issue if it is important to you.

**Comment by Sergio1C** [comment]
is there any news on this?

**Comment by ajcvickers** [comment]
@Sergio1C Nothing has changed since my previous comment.

**Comment by TWhidden** [comment]
On the side note - Postgres allows streaming from their large object store. We store some images, and when we serve it out via our controllers, its EF stream object -> Controller stream, without major allocations.  I'm just here to find how to do it with SQL Server. Guess it's a byte[] / blob today.

**Comment by Neme12** [comment]
It's such an ugly (and very memory inefficient) workaround that you have to realize the whole `byte[]`. which could be megabytes in size or more, as a `byte[]` in memory :/ This should really be in the box.

**Comment by Neme12** [comment]
I'm surprised this still isn't in after 8 years. It's an important scenario - there's no other way to deal with (potentially) big data. Heck, even System.Text.Json supports serializing via a `Stream` and their payloads are probably significantly smaller than a typical file.

**Comment by Neme12** [comment]
As a workaround, I have to store the files in the file system... which is, ugly and most importantly, buggy and wrong as there's no way to reliably have a transaction between a database and a file system.
